src/libmongoc/src/mongoc/mongoc-openssl.c

```diff
@@ -702,12 +703,39 @@ _contact_ocsp_responder (OCSP_CERTID *id, X509 *peer)
          GOTO (retry);
       }
 
-      if (!(resp = OCSP_sendreq_bio (bio, path, req))) {
-         MONGOC_DEBUG (
-            "Could not perform an OCSP request for url '%s'. Error: %s",
-            url,
-            ERR_STR);
+      /* Leave OCSP request NULL, set it onto the request context after setting
+       * the host header. */
+      sendreq_ctx =
+         OCSP_sendreq_new (bio, path, NULL /* OCSP request */, 0 /* maxline */);
+      if (host) {
+         if (0 == OCSP_REQ_CTX_add1_header (sendreq_ctx, "Host", host)) {
+            MONGOC_DEBUG ("Could not set OCSP request header for host: %s",
+                          host);
+            GOTO (retry);
+         }
+      }
+
+      if (0 == OCSP_REQ_CTX_set1_req (sendreq_ctx, req)) {
+         MONGOC_DEBUG ("Could not set OCSP request");
+         GOTO (retry);
       }
+
+      do {
+         int ret = OCSP_sendreq_nbio (&resp, sendreq_ctx);
```

Comment:
`OCSP_sendreq_bio` works with an `OCSP_REQUEST`, not an `OCSP_REQ_CTX`. The only API I saw for sending `OCSP_REQ_CTX` was the non-blocking send, so this now accounts for retries.